crates/driver/src/domain/competition/mod.rs

```diff
@@ -122,43 +126,14 @@ impl Competition {
             })
             .collect_vec();
 
-        // TODO(#1483): parallelize this
-        // TODO(#1480): more optimal approach for settlement merging
-
-        // Merge the settlements in random order.
-        settlements.shuffle(&mut rand::thread_rng());
-
-        // The merging algorithm works as follows: the [`settlements`] vector keeps the
-        // "most merged" settlements until they can't be merged anymore, at
-        // which point they are moved into the [`results`] vector.
-
-        // The merged settlements in their final form.
-        let mut results = Vec::new();
-        while let Some(settlement) = settlements.pop() {
-            // Has [`settlement`] been merged into another settlement?
-            let mut merged = false;
-            // Try to merge [`settlement`] into some other settlement.
-            for other in settlements.iter_mut() {
-                match other.merge(&settlement, &self.eth, &self.simulator).await {
-                    Ok(m) => {
-                        *other = m;
-                        merged = true;
-                        observe::merged(&settlement, other);
-                        break;
-                    }
-                    Err(err) => {
-                        observe::not_merged(&settlement, other, err);
-                    }
-                }
-            }
-            // If [`settlement`] can't be merged into any other settlement, this is its
-            // final, most optimal form. Push it to the results.
-            if !merged {
-                results.push(settlement);
-            }
-        }
-
-        let settlements = results;
+        // Merge settlements
+        merge_settlements(
+            &mut settlements,
+            &self.eth,
+            &self.simulator,
+            auction.deadline(),
```

Comment:
Looks like the architecture was designed such that there should be no need for cloning the auction. Instead of cloning the whole thing we could also just clone the deadline.